ludwig/trainers/trainer.py

```diff
@@ -480,7 +480,15 @@ def tune_batch_size(
 
             def _is_valid_batch_size(batch_size):
                 # make sure that batch size is valid (e.g. less than size of ds)
-                return batch_size < len(training_set)
+                is_smaller_than_training_set = batch_size < len(training_set)
+                is_under_max_batch_size = self.max_batch_size is None or batch_size < self.max_batch_size
```

Comment:
nit: Looks like `batch_size == max_batch_size` is an invalid value, which seems counter-intuitive. I'd assume that the valid values of `batch_size` would be `<= max_batch_size`

Reply:
Good catch. Part of the rationale of this in the first place was that, on the Predibase side, `max_batch_size` was set to `prev_rendered_config['trainer']['batch_size']`. This meant that you needed a strict `<` to limit `tune_batch_size`'s search to values strictly less than the previous attempt's batch size.

I made the change here, and ensured that `max_batch_size` was manually set to 1/2 on the Predibase side.